src/users/users.service.ts

```diff
@@ -56,6 +56,20 @@ export class UsersService {
     return `This action updates a #${id} user`;
   }
 
+  async updateStatus(id: UUID) {
+    try {
+      const user = await this.usersRepository.findOne({
+        where: { id },
+      });
+      if (!user) throw new NotFoundException('User not found');
```

Comment:
This approach is okay as long as it works, but you should take a look at what [TypeORM offers](https://typeorm.io/decorator-reference#deletedatecolumn) out-of-the-box to implement soft-deletable entities